src/core/builders/create-form-validator.ts

```diff
@@ -90,8 +90,10 @@ export const createFormValidator = <Values extends object, Err>(
         const dependents = flatMap(fields, x =>
           getDependents(x, dependenciesDict, getValue)
         );
+        const parents = flatMap([...allFields, ...dependents], getParentsChain);
```

Comment:
It's hard to reason about it but my intuition tells me that it would be sufficient to resolve parents based only on input fields

```ts
const parents = flatMap(fields, getParentsChain)

```

a) the need for validating parents is driven by the change to the value of children (meaning the value of parent is also changed)

b) dependent validation is a separate case - parent of dependent field does not need to be valdiated again as the dependend field value didn't change

c) getting parents of children seems redundant

worth investigating as this would improve perf